asdf/_node_info.py

```diff
@@ -5,7 +5,54 @@
 from .treeutil import get_children
 
 
-def collect_schema_info(key, path, node, identifier="root", preserve_list=True, refresh_extension_manager=False):
+def _filter_tree(info, filters):
+    """
+    Remove nodes from the tree that get caught in the filters.
```

Comment:
I find this description confusing. Shouldn't say something more like keep nodes in the tree that pass the filters? Getting caught in a filter is not as clear to me.